lib/serverengine/process_manager.rb

```diff
@@ -204,14 +219,31 @@ def tick(blocking_timeout=0)
         return nil
       end
 
-      ready_pipes, _, _ = IO.select(@rpipes.keys, nil, nil, blocking_timeout)
+      if ServerEngine.windows?
+        @rpipes.each{|r, m|
```

Comment:
By the way, win32-pipe is not stable and hard to use, because there is many bugs.
Then I think we should use UDPSocket for Heartbeat if it is allowed.